inc/cobs.h

```diff
@@ -9,8 +9,29 @@ namespace cobs {
 
     size_t max_decoded_length(size_t encoded_length);
 
-    size_t encode(const uint8_t * const src_start, size_t src_len, uint8_t * const dst_start, size_t dst_len);
+    struct EncodeResult {
+        enum Status {
```

Comment:
you can change this to enum class to prevent implicit casting to the underlying type
http://en.cppreference.com/w/cpp/language/enum